core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -338,6 +338,130 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
       case Invalid(e) => raiseError(e)
       case Valid(a)   => pure(a)
     }
+
+  /**
+   * Equivalent of a [[scala.Predef#assert(assertion:Boolean,message:=>Any)*]] assertion in `F[_]`.
+   *
+   * Tests an expression, sequencing an exception of type [[java.lang.AssertionError]] if false.
+   *
+   * Assertions can be disabled by using the scalac option `-Xdisable-assertions`, or by
+   * configuring the scalac option `-Xelide-below` with a value greater than
+   * [[scala.annotation.elidable.ASSERTION ASSERTION]].
+   *
+   * When assertions are disabled exceptions are not sequenced, and this method is equivalent to
+   * [[cats.InvariantMonoidal#unit]]
+   *
+   * @param assertion
+   *   the expression to test
+   * @param message
+   *   a String to include in the failure message
+   */
+  final def assert(assertion: Boolean, message: => String)(implicit ev: Throwable <:< E): F[Unit] =
```

Comment:
this seems a bit weird to me. `Predef.assert` to me should be like a rust panic. It shouldn't be a recoverable error. It should represent conditions that should always be true, but we can't (or haven't yet) found a way to encode in the type system.

To lift them into `F[_]` is bringing them into the runtime semantics in a way that I think muddies their purpose

Reply:
Yeah I actually expressed a similar view on typelevel/cats-effect#3044 myself, but raised the PR anyhow for the discussion 😛